#160618120 Reading stats of articles

VictorUkafor opened the pull request:
### What does this PR do?
This pull request allows shows the reading stats of articles

### Description of Task to be completed?

- Refactor method for fetching a single article to save reader's details

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
#160618120

### Screenshots (if appropriate)
N/A

### Questions:
None

AdinoyiSadiq commented on controllers/articleController.js:
```diff
@@ -165,11 +166,45 @@ const reactToArticle = async (req, res) => {
+    error: 'Internal server error'
+  });
+};
+
```
Hi Victor, this looks great. However, I think we should use the 'async await' pattern whenever possible. Also, I would suggest using the 'try and catch' pattern as well.

AdinoyiSadiq commented on controllers/articleController.js:
```diff
@@ -165,11 +166,45 @@ const reactToArticle = async (req, res) => {
+  }
+  const postedRating = await helper.addRating(req.userRating, req.userId, req.article.id);
+  return addRatingsToArticle(res, req, postedRating);
+};
```
Hi Victor. Firstly I like how succinct this function is.

I have a suggestion though. I think addRatingsToArticle functionality shouldn't have 'res, req' passed into it as arguments. From experience, this might lead to errors with node where it tries to send a response twice. 

I would suggest having the function return a value which is then tested by a conditional statement, before returning a response. 

TLDR All responses should be sent from your controller in my opinion.

Commit:
feat(reading-stats): add for displaying reading stats

- add method for saving the number of times user read an article

[Delivers #160618120]

Commit:
feat(reading-stats): add more test cases

- add more test to increase test coverage

[Delivers #160618120]

Commit:
Merge ranch 'staging' into feature/160618120/users-reading-stats

Commit:
Merge branch 'staging' into feature/160618120/users-reading-stats

Commit:
feat(reading-stats): finish implementing the reading stats feature

- refactor logic to fetch all necessary data from models

[Delivers #160618120]

Commit:
feat(reading-stats): fix merge conflict

- fix merge conflict

[Delivers #160618120]

Commit:
feat(reading-stats): fix merge conflict

- fix merge conflict

[Delivers #160618120]

VictorUkafor committed:
Merge branch 'staging' into feature/160618120/users-reading-stats

VictorUkafor committed:
Merge branch 'staging' into feature/160618120/users-reading-stats

VictorUkafor committed:
Update userController.js

Commit:
feat(reading-stats): fix merge conflict

- fix merge conflict

[Delivers #160618120]
